snippets/base/models.py

```diff
@@ -135,13 +137,17 @@ class SnippetBundle(object):
     def __init__(self, client):
         self.client = client
         self._snippets = None
+        self._key = None
 
     @property
     def key(self):
         """A unique key for this bundle as a sha1 hexdigest."""
         # Key should consist of snippets that are in the bundle plus any
         # properties of the client that may change the snippet code
         # being sent.
+        if self._key:
```

Comment:
This works fine, but you could also use [Django's `@cached_property` decorator](https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.functional.cached_property) for these kinds of methods. It basically runs the function the first time, then alters the class so that the method is replaced by a simple property with the value of the function's return. Not a necessary change at all, but cool I think for future reference.

Reply:
Useful, thanks!